pandas/_libs/parsers.pyx

```diff
@@ -178,13 +178,13 @@ cdef extern from "parser/tokenizer.h":
         char thousands
 
         int header # Boolean: 1: has header, 0: no header
-        int header_start # header row start
-        int header_end # header row end
+        ssize_t header_start # header row start
+        ssize_t header_end # header row end
```

Comment:
In some specific places, `-1` is used as a sentinel value. This patch _should_ exhaustively identify all of those cases or change them to not need to use a negative number.